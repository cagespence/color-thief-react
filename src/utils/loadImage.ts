/**
 * Load a image in a promise
 */
export default function loadImage(
  url: string,
  crossOrigin: string | null = null
): Promise<HTMLImageElement> {
  return new Promise((resolve, reject) => {
    const img = new Image();
    img.addEventListener('load', () => {
      resolve(img);
    });
    img.addEventListener('error', () => {
      reject(new Error(`Failed to load image's URL: ${url}`));
    });
    img.crossOrigin = crossOrigin;
    img.src = url;
  });
}
