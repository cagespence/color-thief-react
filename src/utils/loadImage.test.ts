import loadImage from './loadImage';

describe('loadImage', () => {
  test('should try to load a image and load correctly', async () => {
    const img = 'https://homepages.cae.wisc.edu/~ece533/images/airplane.png';
    await expect(loadImage(img)).resolves.toMatchSnapshot();
  });
  test('should try load a false image and throw an error', async () => {
    const img = 'https://homepages.cae.wisc.edu/~ece533/images/airplane.jpg';
    await expect(loadImage(img)).rejects.toThrow(Error);
  });
});
