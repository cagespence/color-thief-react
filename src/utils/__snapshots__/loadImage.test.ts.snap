// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadImage should try to load a image and load correctly 1`] = `
<img
  crossorigin="null"
  src="https://homepages.cae.wisc.edu/~ece533/images/airplane.png"
/>
`;
